File: java/eyes.sdk.core/src/main/java/com/applitools/eyes/universal/Refer.java
package com.applitools.eyes.universal;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;

public abstract class Refer {

    public static final String APPLITOOLS_REF_ID = "applitools-ref-id";

    protected Map<String, Object> references;
    protected Map<String, HashSet<Reference>> relations;

    public Refer() {
        references = new HashMap<>();
        relations = new HashMap<>();
    }

    protected abstract void destroy(Reference root);
}

File: dotnet/Eyes.Images.DotNet/Universal/Refer.cs
using System.Collections.Generic;

namespace Applitools.Universal
{
    public abstract class Refer
    {
        public static readonly string APPLITOOLS_REF_ID = "applitools-ref-id";

        protected Dictionary<string, object> References = new Dictionary<string, object>();
        protected Dictionary<string, HashSet<Reference>> Relations = new Dictionary<string, HashSet<Reference>>();

        public abstract void Destroy(Reference root);
    }
}
